File: src/main/java/com/ism/dreamssellersv2/service/VendorService.java
package com.ism.dreamssellersv2.service;

import com.ism.model.VendorCreateDTO;
import com.ism.model.VendorDTO;

import java.util.List;

public interface VendorService {
    List<VendorDTO> getVendors();
    VendorDTO getVendorById(Integer vendorId);
    VendorDTO createVendor(VendorCreateDTO vendorCreateDTO);
    void deleteVendor(Integer vendorId);
}

File: frontend/src/app/services/vendor.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { VendorDTO, VendorCreateDTO } from '../models/vendor.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class VendorService {
  private apiUrl = `${environment.apiUrl}/vendors`;

  constructor(private http: HttpClient) {}

  createVendor(vendor: VendorCreateDTO): Observable<VendorDTO> {
    return this.http.post<VendorDTO>(this.apiUrl, vendor);
  }

  getVendors(): Observable<VendorDTO[]> {
    return this.http.get<VendorDTO[]>(this.apiUrl);
  }

  getVendorById(vendorId: number): Observable<VendorDTO> {
    return this.http.get<VendorDTO>(`${this.apiUrl}/${vendorId}`);
  }

  deleteVendor(vendorId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${vendorId}`);
  }
}
